sample/src/main/java/com/google/modernstorage/sample/mediastore/MediaStoreViewModel.kt

```diff
@@ -21,129 +21,113 @@ import androidx.lifecycle.AndroidViewModel
-            }
-        }
-    }
+    val currentFile: LiveData<FileResource> = savedStateHandle.getLiveData(CURRENT_MEDIA_KEY)
 
-    fun setCurrentMedia(uri: Uri) {
-        viewModelScope.launch {
-            mediaStore.getResourceByUri(uri)?.let {
-                savedStateHandle.set("currentMediaUri", uri)
-                _currentMedia.value = mediaStore.getResourceByUri(uri)
-            }
-        }
+    suspend fun setCurrentMedia(uri: Uri) {
+        savedStateHandle.set(CURRENT_MEDIA_KEY, mediaStore.getResourceByUri(uri).getOrNull())
     }
 
     val temporaryCameraImageUri: Uri?
-        get() = savedStateHandle.get("temporaryCameraImageUri")
+        get() = savedStateHandle.get(TEMPORARY_CAMERA_IMAGE_URI_KEY)
 
     fun saveTemporaryCameraImageUri(uri: Uri) {
-        savedStateHandle.set("temporaryCameraImageUri", uri)
+        savedStateHandle.set(TEMPORARY_CAMERA_IMAGE_URI_KEY, uri)
     }
 
     fun clearTemporaryCameraImageUri() {
-        savedStateHandle.remove<Uri>("temporaryCameraImageUri")
+        savedStateHandle.remove<Uri>(TEMPORARY_CAMERA_IMAGE_URI_KEY)
     }
 
-    fun saveRandomImageFromInternet(callback: (uri: Uri) -> Unit) {
-        viewModelScope.launch {
-            val request = Request.Builder().url(SampleData.image.random()).build()
-
-            withContext(Dispatchers.IO) {
-                val response = httpClient.newCall(request).execute()
-
-                response.body?.use { responseBody ->
-                    val filename = generateFilename(MediaSource.INTERNET, "jpg")
-                    val imageUri = mediaStore.addImageFromStream(
-                        filename = filename,
-                        inputStream = responseBody.byteStream(),
-                        location = SharedPrimary
-                    )
+    suspend fun saveRandomMediaFromInternet(type: FileType): Result<FileResource> {
+        val url: String
+        val extension: String
+        val mimeType: String
 
-                    withContext(Dispatchers.Main) {
-                        callback(imageUri)
-                    }
-                }
+        when (type) {
+            FileType.IMAGE -> {
+                url = SampleData.image.random()
+                extension = "jpg"
+                mimeType = "image/jpeg"
+            }
+            FileType.VIDEO -> {
+                url = SampleData.video.random()
+                extension = "mp4"
+                mimeType = "video/mp4"
             }
+            else -> throw IllegalArgumentException("Unsupported type: $type")
         }
-    }
 
-    fun saveRandomVideoFromInternet(callback: (uri: Uri) -> Unit) {
-        viewModelScope.launch {
-            val request = Request.Builder().url(SampleData.video.random()).build()
+        val request = Request.Builder().url(url).build()
 
-            withContext(Dispatchers.IO) {
-                val response = httpClient.newCall(request).execute()
+        return withContext(Dispatchers.IO) {
```

Comment:
```suggestion
        return withContext(ioDispatcher) {
```

The IO dispatcher should be injected into the ViewModel to make testing easier